numpy/core/src/common/npy_partition.h

```diff
@@ -0,0 +1,27 @@
+#ifndef __NPY_PARTITION_H__
+#define __NPY_PARTITION_H__
```

Comment:
I'm going to need some help understanding the mechanism here. I'm guessing the `taglist` is key along with the `get_*` functions.